Mathlib/RingTheory/LocalProperties/LinearMap.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2024 Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sihan Su, Yongle Hu, Yi Song
+-/
+import Mathlib.RingTheory.LocalProperties.Submodule
+import Mathlib.Algebra.Exact
+/-!
+# Local properties about linear maps
+
+In this file, we show that injective, surjective, bijective (of linear maps) are local properties.
+
+-/
+open Submodule LocalizedModule Ideal LinearMap
+
+section localization_maximal
+
+variable {R M N : Type*} [CommRing R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
+  (f : M →ₗ[R] N)
+
+theorem injective_of_localization
+    (h : ∀ (J : Ideal R) (_ : J.IsMaximal), Function.Injective (map J.primeCompl f)) :
+    Function.Injective f :=
+  ker_eq_bot.mp <| eq_bot_of_localization_maximal' <|
+  fun J hJ ↦ (localized'_ker_eq_ker_localizedMap _ _ _ _ f).trans <| ker_eq_bot.mpr <| h J hJ
```

Comment:
Could you please formulate all these lemmas for an arbitrary family of module satisfying the correct `IsLocalizedModule` predicate? You can have a look at `LocalProperties.Submodule` for a possible setup. I think it should not cause significantly more work, since all the results you use are already formulated in the general setup.